### PR TITLE
Filtering Rules Implementation

### PR DESCRIPTION
**Pull request by yoouung**
: working on the implementation on frontend details and backend connection 

### Frontend
- add details on rules.js
- add `axiosInterceptor` in `components/hooks/api/` to process axios response
- delete `keywords` option temporarily
- add `proxy` in `package.json`

### Backend
- set `args`' form on app.py

### CODE REVIEW

**Comment by yoouung** [comment]
### 1. Selecting option
image

### 2. Alert for success
image

### 3. Result
`original data`
image
`filtered data`
image

File: dt-demo-be/app.py [hunk 1]
```diff
@@ -28,10 +28,19 @@ def filter():
             modules="dup",
             input="demo.txt"  
         )
-        args = sampleArgs1
+        # args = sampleArgs1
```
Comment (doorooful) [inline dt-demo-be/app.py:31]: Please remove unnecessary lines and replace them into comments!

File: dt-demo-be/app.py [hunk 2]
```diff
@@ -28,10 +28,19 @@ def filter():
...
+        args = modules.argparse.Namespace(
+            all=data['form']['all'],
+            modules=data['form']['modules'],
+            input=data['form']['input']
+        )
+
         app.logger.info(f"arguments: all={args.all}, modules={args.modules}, input={args.input}")
         modules.runModule(args)
-        return "POST return"
+        return "Filtering done"
```
Comment (doorooful) [inline dt-demo-be/app.py:43]: I think `"filtering requested"` would be better

File: dt-demo-fe/package.json [hunk 3]
```diff
@@ -2,12 +2,15 @@
     "name": "dt-demo-fe",
     "version": "0.1.0",
     "private": true,
+    "proxy": "http://localhost:5001",
```
Comment (doorooful) [inline dt-demo-fe/package.json:5]: Could you test this line?

File: dt-demo-fe/src/components/Rules/ruleOptions.js [hunk 4]
```diff
@@ -0,0 +1,30 @@
...
+        value: "dup",
+        description: "dup 모듈에 대한 설명입니다.",
+    },
+    {
+        label: "개인정보",
+        value: "pdd",
+        description: "pdd 모듈에 대한 설명입니다.",
+    },
+
+    // {
```
Comment (doorooful) [inline dt-demo-fe/src/components/Rules/ruleOptions.js:23]: Please remove not used lines!

**Review by doorooful — CHANGES_REQUESTED** [review]
I request changes according to the comments ◡̈

File: dt-demo-fe/package.json [hunk 5]
```diff
@@ -7,7 +7,9 @@
         "@testing-library/jest-dom": "^5.17.0",
         "@testing-library/react": "^13.4.0",
         "@testing-library/user-event": "^13.5.0",
+        "antd": "^5.9.0",
         "axios": "^1.5.0",
+        "fs": "^0.0.1-security",
```
Comment (doorooful) [inline dt-demo-fe/package.json:12]: Can you check if npm installing command is needed?

**Review by doorooful — CHANGES_REQUESTED** [review]
Please find the comment!!

**Review by doorooful — APPROVED** [review]
LGTM!!

**Comment by doorooful** [comment]
Please use following sample data as follows
```JSON
{
    "form": {
        "all": false,
        "modules": "typo,pdd",
        "input": "demo.txt"
    }
}
```

image

**Pull request merged**